interfaces/network-control: Allow systemd resolved cache flushing via D-Bus

jocado opened the pull request:
Add the `FlushCaches` method to the already existing rules that allow resolved management. The existing access is for safe members of the systemd-resolved D-Bus API, and FlushCache should also be a safe operation.

Some utilities that configure systemd resolved via D-Bus use flush caches as part of their setup flow.

One such use case is Tailscale, which this changes enables use of in a strictly confined snap. Discussed here: https://github.com/tailscale/tailscale/issues/267

bboozzoo reviewed (approved):
LGTM

zyga reviewed (approved):
Looks good

Apparmor does not have a profile for systemd/resolved.

alexmurray reviewed (approved):
LGTM but it would perhaps be good to unify all the various rules in network-control for resolve1 since one lot specifies:

```
peer=(name="org.freedesktop.resolve1", label=unconfined),
```

whilst the other has just

```
peer=(label=unconfined),
```

and so now we will have 2 using just the label and a third with the additional name restriction. Ideally all 3 would be consistent.

jocado commented:
Thanks for review @alexmurray 

Two things:

- I've added harmonised the `peer` across all rules with the extra name restriction.  That seems to work fine :+1: 
- I've also added a rule to allow `org.freedesktop.DBus.Peer.Ping` for resolved, as I noticed that the use case I'm testing for also makes us of that.  I believe it's a standard DBus client behaviour to use that.  I've added it as a separate commit for now, pending your feedback.  I will squash it if it's ok, or remove it it's not.

alexmurray commented:
Thanks @jocado - looks great.

jocado commented:
Thanks @alexmurray , I have squashed the commits together.

Ready for final review/merge whenever someone can please.  Thanks!
